### ISSUE TITLE
Printing of a Permutation in array form is wrong

### ISSUE THREAD

**Issue by gxyd** [sympy/sympy#11482]
```
>>> from sympy import *
>>> from sympy.combinatorics.permutations import Permutation
>>> Permutation.print_cyclic = False
>>> p = Permutation([0, 2, 1])
>>> p
Permutation[0, 2, 1] # wrong printing
```

**Comment by ArifAhmed1995** [comment]
@gxyd It should be printed as a list ? Like this : `[0, 2, 1]` ?

**Comment by ArifAhmed1995** [comment]
@gxyd It should be printed as a list ? Like this : `[0, 2, 1]` ?

**Comment by ArifAhmed1995** [comment]
@gxyd It should be printed as a list ? Like this : `[0, 2, 1]` ?

**Comment by gxyd** [comment]
No. I am sure it shouldn't be printed like a list. I will have to look 
as to what should be the correct printing, currently my exams are going on.

**Comment by mohakuma** [comment]
I want to work on this issue!

**Comment by ArifAhmed1995** [comment]
@mohakuma Sure. Go for it.

**Comment by karan1276** [comment]
@gxyd how should this actually be printed?

**Comment by gxyd** [comment]
It should be printed as:
```
>>> from sympy import *
>>> from sympy.combinatorics.permutations import Permutation
>>> Permutation.print_cyclic = False
>>> p = Permutation([0, 2, 1])
>>> p
Permutation([0, 2, 1])
```
i.e it should be printed as it is.
Apart from that, I think the whole [code block here](https://github.com/sympy/sympy/blob/master/sympy/combinatorics/permutations.py#L982-L1004) is untested. So it would be really good if it could be tested completely, perhaps to make sure it is untested you should first use some code-coverage tool first.

**Comment by karan1276** [comment]
@gxyd i didn't get you, what code coverage tools are you talking about? is writing python tests an option?

**Comment by gxyd** [comment]
> is writing python tests an option?

No for the changes required for the current issue, writing tests is mandatory.
> what code coverage tools are you talking about?

If you ever want to get the report, just run:
```
gaurav@stallman:~/OpenSource/sympy$ bin/coverage_report.py sympy/combinatorics/
You need to install module coverage (version 3.4 or newer required).
See http://nedbatchelder.com/code/coverage/ or 
https://launchpad.net/ubuntu/+source/python-coverage/
```
and click on the file to know about the untested files. Though I have not installed coverage tools, so I am referred to installing the code-coverage package. After you install that you can have a coverage report like [this](https://cloud.githubusercontent.com/assets/5647941/8894100/38b38d76-33c8-11e5-8e87-fc18e54a6a48.png) though in a browser.
